src/CLI/OutputGenerator.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Axe.Windows.Automation;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace AxeWindowsScanner
+{
+    public class OutputGenerator : IOutputGenerator
+    {
+        private readonly TextWriter _writer;
+
+        private bool _bannerHasBeenShown;
+
+        public OutputGenerator(TextWriter writer)
+        {
+            if (writer == null) throw new ArgumentNullException(nameof(writer));
+
+            _writer = writer;
+        }
+
+        public void ShowOutput(IOptions options, IErrorCollector errorCollector, ScanResults scanResults)
+        {
+            bool failedToComplete = errorCollector.ParameterErrors.Any() || errorCollector.Exceptions.Any() || (scanResults == null);
```

Comment:
Could this logic be part of the ErrorCollector? It might make the code a little more succinct.

Reply:
Wrapped this in IErrorCollector.Any